File: 6.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
 
int main() {
    char enter[6];
    scanf("%s", enter);
     
    if (strlen(enter) != 5 || enter[2] != '-') {
        printf("ERROR\n");
        exit(0);
    }
     
    char check[] = "12345678ABCDEFGH-";
    for (int i = 0; i < strlen(enter); i++) {
        if (strchr(check, enter[i]) == NULL) {
            printf("ERROR\n");
            exit(0);
        }
    }
     
    char one[3], two[3];
    sscanf(enter, "%2s-%2s", one, two);
     
    if (strlen(enter) == 5 && strlen(one) == 2 && strlen(two) == 2) {
        if (65 <= one[0] && one[0] <= 90 && 65 <= two[0] && two[0] <= 90 && 48 <= one[1] && one[1] <= 57 && 48 <= two[1] && two[1] <= 57) {
            if (two[0] == one[0] - 2 && (two[1] == one[1] - 1 || two[1] == one[1] + 1)) {
                printf("YES\n");
            } else if (two[0] == one[0] + 1 && (two[1] == one[1] - 2 || two[1] == one[1] + 2)) {
                printf("YES\n");
            } else if (two[0] == one[0] + 2 && (two[1] == one[1] - 1 || two[1] == one[1] + 1)) {
                printf("YES\n");
            } else if (two[0] == one[0] - 1 && (two[1] == one[1] - 2 || two[1] == one[1] + 2)) {
                printf("YES\n");
            } else {
                printf("NO\n");
            }
        } else {
            printf("ERROR\n");
            exit(0);
        }
    } else {
        printf("ERROR\n");
        exit(0);
    }
     
    return 0;
}

File: 6.py
enter = input()
 
if len(enter)!=5 or enter[2] != '-':
    print("ERROR")
    exit(0)
 
check = ['1','2','3','4','5','6','7','8','A','B','C','D','E','F','G','H','-']
for i in enter:
    if i not in check:
        print("ERROR")
        exit(0)
 
one, two = enter.split("-")
if len(enter)==5 and len(one)==2 and len(two)==2:
    if 65 <= ord(one[0]) <= 90 and 65 <= ord(two[0]) <= 90 and 48 <= ord(one[1]) <= 57 and 48 <= ord(two[1]) <= 57:
        if two[0] == chr(ord(one[0])-2) and (two[1] == chr(ord(one[1])-1) or two[1] == chr(ord(one[1])+1)):
                print("YES")
        elif two[0] == chr(ord(one[0])+1) and (two[1] == chr(ord(one[1])-2) or two[1] == chr(ord(one[1])+2)):
                print("YES")
        elif two[0] == chr(ord(one[0])+2) and (two[1] == chr(ord(one[1])-1) or two[1] == chr(ord(one[1])+1)):
                print("YES")
        elif two[0] == chr(ord(one[0])-1) and (two[1] == chr(ord(one[1])-2) or two[1] == chr(ord(one[1])+2)):
                print("YES")
        else:
            print("NO")
    else:
        print("ERROR")
        exit(0)
else:
    print("ERROR")
    exit(0)
